pkg/local/sync.go

```diff
@@ -87,51 +87,50 @@ func NewSyncManager(ctx context.Context, client *api.ClientWithResponses, maxPar
 	}
 }
 
+// Sync - sync changes between remote and local directory given the Changes channel.
+// For each change, will apply download, upload or delete according to the change type and change source
 func (s *SyncManager) Sync(rootPath string, remote *uri.URI, changeSet <-chan *Change) error {
```

Comment:
I have a confession:
I asked chatGPT if he can improve the documentation. Expected it to make slight changes, got a bit more than 
that, but I had to put it out here:

```
// Sync synchronizes changes between a local directory and a remote URI.
// The function consumes changes provided through the changeSet channel.
// Depending on the type and source of each change, the Sync function will
// either download, upload, or delete the associated content.
// 
// Parameters:
// - rootPath: The local directory path that should be kept in sync with the remote.
// - remote: The URI representing the remote directory.
// - changeSet: A channel through which changes to be synchronized are sent.
//
// The function leverages goroutines for parallel processing of changes, and the 
// maximum number of parallel operations is determined by s.maxParallelism. The synchronization 
// process is stopped early if an error is encountered.
// 
// After syncing, any empty directories within rootPath are pruned.
// 
// Returns:
// - nil if the sync completes successfully.
// - An error if any operation during the sync or pruning process fails.
```
It isn't aligned with our documentation, so I'm not sure what I'd suggest, but still needed to share 🤷